tor/core/user_interaction.py

```diff
@@ -74,39 +73,45 @@ def process_claim(post, config):
-        # do not cache this page. We want to get it every time.
-        post.reply(_(
-            please_accept_coc.format(get_wiki_page('codeofconduct', config))
-        ))
-        return
+    try:
+        if not coc_accepted(post, config):
+            # do not cache this page. We want to get it every time.
+            post.reply(_(
+                please_accept_coc.format(get_wiki_page('codeofconduct', config))
+            ))
+            return
+
+        if top_parent.link_flair_text is None:
+            # There exists the very small possibility that the post was malformed
+            # and doesn't actually have flair on it. In that case, let's set
+            # something so the next part doesn't crash.
+            flair_post(top_parent, flair.unclaimed)
+
+        if flair.unclaimed in top_parent.link_flair_text:
+            # need to get that "Summoned - Unclaimed" in there too
+            post.reply(_(claim_success))
 
-    if top_parent.link_flair_text is None:
-        # There exists the very small possibility that the post was malformed
-        # and doesn't actually have flair on it. In that case, let's set
-        # something so the next part doesn't crash.
-        flair_post(top_parent, flair.unclaimed)
+            flair_post(top_parent, flair.in_progress)
+            logging.info(
+                'Claim on ID {} by {} successful'.format(
+                    top_parent.fullname, post.author
+                )
+            )
 
-    if flair.unclaimed in top_parent.link_flair_text:
-        # need to get that "Summoned - Unclaimed" in there too
-        try:
-            post.reply(_(claim_success))
-        except:
-            # Pause bubbling up the stack trace to log this error
-            logging.error('ERROR: {1} of type {0}\n{2}'.format(*sys.exc_info()), extra={'stack': True})
-            raise  # Bubble up the original exception
-
-        flair_post(top_parent, flair.in_progress)
-        logging.info(
-            'Claim on ID {} by {} successful'.format(
-                top_parent.fullname, post.author
+        # can't claim something that's already claimed
+        elif top_parent.link_flair_text == flair.in_progress:
+            post.reply(_(already_claimed))
+        elif top_parent.link_flair_text == flair.completed:
+            post.reply(_(claim_already_complete))
+
+    except praw.exceptions.APIException as e:
+        if e.error_type == 'DELETED_COMMENT':
+            logging.debug(
+                'Comment attempting to claim ID {} has been deleted. '
+                'Back up for grabs!'.format(top_parent.fullname)
             )
-        )
-    # can't claim something that's already claimed
-    elif top_parent.link_flair_text == flair.in_progress:
-        post.reply(_(already_claimed))
-    elif top_parent.link_flair_text == flair.completed:
-        post.reply(_(claim_already_complete))
+            return
+        raise  # Re-raise exception if not
```

Comment:
Does `raise` in this context actually reraise the exception with all its information? Seems like we would want to raise `e`.

Reply:
I thought the same, but:

> When raising (or re-raising) an exception in an except or finally clause __context__ is automatically set to the last exception caught; if the new exception is not handled the traceback that is eventually displayed will include the originating exception(s) and the final exception.

(from [python docs on exceptions](https://docs.python.org/3/library/exceptions.html))

Reply:
Huh, I sit corrected. Nice!

Reply:
In other languages, calling `raise e` like that would clobber the stack trace while `raise` wouldn't. In python it doesn't appear to clobber in the same way like I originally thought. In any event, I left a comment to clarify what's happening here. :smile: